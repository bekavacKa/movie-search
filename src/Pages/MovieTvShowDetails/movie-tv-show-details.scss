@import '../../Assets/scss/variables.scss';
@import '../../Assets/scss/mixins.scss';

.movie-tv-show-details{
    // background-color: $secondBackColor;
    // padding: 25px 50px;
    @include my-flex(center, column, center);
    .details-back{
        @include my-container;
        // background-color: blue;
        padding: 30px 5px;
        @include my-flex(flex-start, row, center);
        align-self: flex-start;
        transition: $transition;
        cursor: pointer;
        &:hover{
            transform: scale(1.01);
        }
        .details-back-icon{
            font-size: 2rem;
        }
        .details-back-span{
            margin: 0 20px;
            font-size: 1.2rem;
        }
    }
    .details-content{
        width: 100%;
    }
}

@media only screen and (max-width: 319px) {
    .movie-tv-show-details{
        .details-back{
            padding: 10px 5px;
        }
    }
}

@media only screen and (min-width: 320px) and (max-width: 768px){
    .movie-tv-show-details{
        .details-back{
            padding: 15px 5px;
        }
    }
}

@media only screen and (min-width: 769px) and (max-width: 1024px){
    .movie-tv-show-details{
        .details-back{
            padding: 20px 5px;
        }
    }
}