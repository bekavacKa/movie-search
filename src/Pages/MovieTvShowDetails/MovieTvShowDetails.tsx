import React from 'react';
import { Link } from 'react-router-dom';
import { FaAngleLeft } from 'react-icons/fa';
import './movie-tv-show-details.scss';
import routes from '../../Config/routes';

function MovieTvShowDetails() {
  return (
    <div className='movie-tv-show-details'>

        <Link to={routes.HOME.url} className='details-back'>
            <FaAngleLeft className='details-back-icon'/> 
            <span className='details-back-span'> BACK </span>
        </Link>

        <div className='details-image'>
            <img src='https://live.staticflickr.com/4005/4686746190_31c292dc8c_b.jpg' alt='slika' />
        </div>

        <div className='details-title'>
            <h2>
                MovieTvShowDetails
            </h2>
        </div>

        <div className='details-description'>
            <p>
                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
            </p>
        </div>

    </div>
  )
}

export default MovieTvShowDetails