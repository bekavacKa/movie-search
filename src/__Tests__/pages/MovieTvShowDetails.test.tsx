import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';
import MovieTvShowDetails from '../../Pages/MovieTvShowDetails/MovieTvShowDetails';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

test('Snapshot test', () => {
    const initialState = { 
        searchTermStore: '',
        buttonsStore:{
            tvShowsButton: true,
            movieButton: false
        },
        storeSearchTerm: ""
    };
    const mockStore = configureStore();
    let store;

    store = mockStore(initialState);
  const tree = renderer.create(
    <Provider store={store}>
        <Router>
            <MovieTvShowDetails />
        </Router>
    </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
