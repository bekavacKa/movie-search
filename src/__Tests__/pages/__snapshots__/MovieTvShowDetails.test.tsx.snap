// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<div
  className="movie-tv-show-details"
>
  <a
    className="details-back"
    href="/"
    onClick={[Function]}
  >
    <svg
      className="details-back-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 256 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
      />
    </svg>
    <span
      className="details-back-span"
    >
       BACK 
    </span>
  </a>
  <div
    className="details-content"
  >
    <div
      className="card-details"
      style={
        {
          "backgroundImage": "linear-gradient( rgba(0, 0, 0, 0.9), rgba(0, 0, 0, 0.9)), url(https://www.themoviedb.org/t/p/w220_and_h330_face/undefined)",
        }
      }
    >
      <div
        className="card-details-image"
      >
        <img
          src="https://www.themoviedb.org/t/p/w220_and_h330_face/undefined"
        />
      </div>
      <div
        className="card-details-info"
      >
        <div
          className="card-details-title"
        >
          <h2 />
        </div>
        <div
          className="card-details-overview"
        >
          <p />
        </div>
      </div>
    </div>
  </div>
</div>
`;
